stock_cubiscan/wizard/cubiscan_wizard.py

```diff
@@ -23,46 +23,90 @@ class CubiscanWizard(models.TransientModel):
-                    "sequence": seq + 1,
-                    "name": pack_type.name,
-                    "qty": 0,
-                    "max_weight": 0,
-                    "lngth": 0,
-                    "width": 0,
-                    "height": 0,
-                    "barcode": False,
-                    "packaging_type_id": pack_type.id,
-                }
-                if pack:
-                    vals.update(
-                        {
-                            "qty": pack.qty,
-                            "max_weight": pack.max_weight,
-                            "lngth": pack.lngth,
-                            "width": pack.width,
-                            "height": pack.height,
-                            "barcode": pack.barcode,
-                            "packaging_id": pack.id,
-                            "packaging_type_id": pack_type.id,
-                        }
-                    )
-                to_create.append(vals)
+            to_create += self._prepare_unit_line()
+            to_create += self._prepare_packaging_lines()
             recs = self.env["cubiscan.wizard.line"].create(to_create)
             self.line_ids = recs
         else:
             self.line_ids = [(5, 0, 0)]
 
+    def _prepare_unit_line(self):
+        vals = {
+            "wizard_id": self.id,
+            "sequence": 0,
+            "name": "Unit",
+            "qty": 1,
+            "max_weight": self.product_id.weight,
+            "lngth": self.product_id.product_length,
+            "width": self.product_id.product_width,
+            "height": self.product_id.product_height,
+        }
+        product_dimension_uom = self.product_id.dimensional_uom_id
+        cubiscan_mm_uom = self.env.ref(
+            "stock_cubiscan.product_uom_mm", raise_if_not_found=False
+        )
+        if not cubiscan_mm_uom:
+            raise exceptions.UserError(
+                _(
+                    "Cannot find 'mm' Unit of measure, please update "
+                    "`stock_cubiscan` module"
+                )
+            )
+        if cubiscan_mm_uom != product_dimension_uom:
+            vals.update(
+                {
+                    "lngth": product_dimension_uom._compute_quantity(
+                        self.product_id.product_length, cubiscan_mm_uom
+                    ),
+                    "width": product_dimension_uom._compute_quantity(
+                        self.product_id.product_width, cubiscan_mm_uom
+                    ),
+                    "height": product_dimension_uom._compute_quantity(
+                        self.product_id.product_height, cubiscan_mm_uom
+                    ),
+                }
+            )
+        return [vals]
+
+    def _prepare_packaging_lines(self):
+        vals_list = []
+        packaging_types = self.env["product.packaging.type"].search([])
+        for seq, pack_type in enumerate(packaging_types):
+            pack = self.env["product.packaging"].search(
```

Comment:
if this wizard is used very often, I'd drop the search and add a read_group by product filtering on all pkg types before the loop.
Then in the loop you can lookup by product id.

Reply:
This part wasn't changed in this PR but only moved to a dedicated function. Sorry but I don't want to mess with its behaviour now...